### PR TITLE
README: Add introduction, attack analysis

### PR DESCRIPTION
**Pull request by b10n**

### CODE REVIEW

**Comment by b10n** [comment]
Changed.

File: README.md [hunk 1]
```diff
@@ -57,15 +56,17 @@ The tool operates in 2 modes - infection and execution.
     - SSH:
         - Connects to the `addr` specified and/or all hosts in `known_hosts` file, using either a running SSH agent(with `SSH_AUTH_SOCK` set in environment) or using a password-less private key, which provides access to the machine(s).
         - Copies the `moosh` binary to each victim using SFTP.
-    - Buffer overflow of a HHTPD server:
+    - Shellshock:
+        - Connects to the `addr`.
+        - Downloads `moosh` binary in `/tmp` and executes it.
+    - Buffer overflow of hHTTPd: (TODO)
```
Comment (rvolosatovs) [inline README.md]: TODO?

File: README.md [hunk 2]
```diff
@@ -4,13 +4,12 @@
...
 
+## Introduction
+Mooshy is a tool that automates the infection and execution of arbitrary code on remote Linux systems. It has various modes of infection. One is by exploiting a web server vulnerable to Shellshock. Another way is by using SSH to login as a regular user. After code execution is gained, the Dirty COW exploit is used to gain root privileges. Using these privileges, a backdoor is installed for persistence.
+
 ## Installation:
 ```sh
     go get -u -v github.com/rvolosatovs/mooshy/cmd/mooshy
-```
-
-## Development:
```
Comment (rvolosatovs) [inline README.md]: Why remove this?

File: README.md [hunk 3]
```diff
@@ -4,13 +4,12 @@
 
 [![asciicast](https://asciinema.org/a/zK5uZERZHIAw1TsNng2o8eDqX.png)](https://asciinema.org/a/zK5uZERZHIAw1TsNng2o8eDqX)
 
+## Introduction
```
Comment (rvolosatovs) [inline README.md]: Description?

File: README.md [hunk 4]
```diff
@@ -85,10 +86,18 @@ The tool operates in 2 modes - infection and execution.
...
 
 ## Backdoor
 The `systemd-timesync` service creates a raw socket that listens to TCP requests. The raw socket allows the backdoor to listen on all TCP ports with a single socket, while the TCP payload is still handled by the kernel so that normal service by the victim machine is unaffected. When a packet with a specific payload is received, it triggers the backdoor to open a reverse shell to the attacker's machine. The payload consists of a sentinel value, followed by the port number on which the attacker is listening for a reverse shell connection.
 
 The backdoor then opens a new TCP connection to the source IP of the triggering packet, using the specified port. Since the backdoor is run as a systemd service, it has not tty attached. Instead, it uses the `script` executable, which is installed on linux by default, to create a pty and connect it to `/bin/bash`. This opens a fully interactive root shell on the victim's machine, whose input and output are then connected to the opened TCP connection. 
+
+## Buffer overflow
```
Comment (rvolosatovs) [inline README.md]: We're not using this anymore, right?

**Pull request merged**